fpga/src/main/scala/arty/Configs.scala

```diff
@@ -25,17 +29,41 @@ class WithDefaultPeripherals extends Config((site, here, up) => {
     idcodeManufId = 0x489,
     debugIdleCycles = 5)
   case SerialTLKey => None // remove serialized tl port
+  case PeripherySPIFlashKey => List(
+    SPIFlashParams(
+      fAddress = 0x20000000,
+      rAddress = 0x10014000,
+      defaultSampleDel = 3))
+  case BootROMLocated(x) => up(BootROMLocated(x), site).map { p =>
+    // invoke makefile for xip
+    val make = s"make -C fpga/src/main/resources/arty/xip bin"
+    require (make.! == 0, "Failed to build bootrom")
+    p.copy(hang = 0x10000, contentFileName = s"./fpga/src/main/resources/arty/xip/build/xip.bin")
+  }
 })
+
 // DOC include start: AbstractArty and Rocket
 class WithArtyTweaks extends Config(
+  new WithArtyGPIOHarnessBinder ++
+  new chipyard.iobinders.WithGPIOIOCells ++
   new WithArtyJTAGHarnessBinder ++
   new WithArtyUARTHarnessBinder ++
   new WithArtyResetHarnessBinder ++
+  new WithArtySPIFlashHarnessBinder ++
+  new WithSPIFlashIOPassthrough ++
   new WithDebugResetPassthrough ++
   new WithDefaultPeripherals ++
+  new chipyard.config.WithPeripheryBusFrequency(32.0) ++
```

Comment:
This line is crucial to Freedom SDK support, as it tells Freedom what frequency the bus is run at (assumes 100MHz if this is not set). Without this line, interfacing to the board using UART gives garbage data/wrong baud rate